### DISCUSSION TITLE
Uncaught TypeError: Cannot read properties of undefined (reading 'admin')

### DISCUSSION THREAD

**Discussion by Dini-25** [mdelobelle/metadatamenu#650] [General]
I have configured Metadata Menu in my obsidian vault using the following approach:

- Added a new folder in my vault called /admin at the root of the vault. 
- Configured Metadata menu settings to use the above folder as the location of the classfiles. 

![image](https://github.com/mdelobelle/metadatamenu/assets/112165865/9de27ba7-9231-44e6-97b0-8c9522f8447f)

- Added a new file in the above folder called Sandbox.md
- Created a new test note that has these two properties

```
status: open
fileClass: Sandbox
```

Everything seem to be working fine as per below:

**Class file:**
![image](https://github.com/mdelobelle/metadatamenu/assets/112165865/f260d744-9884-4428-9887-773a7fdacbb2)

![image](https://github.com/mdelobelle/metadatamenu/assets/112165865/eaffc30a-63e8-4a42-a34a-8004554026d5)

**Test note:**

![image](https://github.com/mdelobelle/metadatamenu/assets/112165865/bd29a44c-6eb2-4cc6-9761-603a9f21323f)

As you can see, the test note is detecting the Class and setting the button/icon correctly. 

The only issue i have is that whenever i have a look at the developer console, i see lots of these errors (the same error being repeated). 

```
app.js:1 Uncaught TypeError: Cannot read properties of undefined (reading 'admin')
    at FileClassFolderButton.addButton (plugin:metadata-menu:25423:44)
    at eval (plugin:metadata-menu:25408:14)
    at e.tryTrigger (app.js:1:726617)
    at e.trigger (app.js:1:726550)
    at t.trigger (app.js:1:2235641)
    at t.updateLayout (app.js:1:2228091)
    at r (app.js:1:260736)
    at node:internal/process/task_queues:140:7
    at AsyncResource.runInAsyncScope (node:async_hooks:204:9)
    at AsyncResource.runMicrotask (node:internal/process/task_queues:137:8)
```

Even though every thing works as expected, i noticed that the above error appears whenever i make a change to any file. 
What could be causing this error?

**Comment by Dini-25** [comment]
Note - I am using the following version

Obsidian
![image](https://github.com/mdelobelle/metadatamenu/assets/112165865/d68cd848-a259-4637-9206-96844c4e6cdd)

Metadata menu
![image](https://github.com/mdelobelle/metadatamenu/assets/112165865/455fbba4-1002-4ed0-b0fb-9a7b1d8f68e9)

**Comment by jjquin** [comment]
I'm having the same issue. I abandoned Metadata Menu when I first started to use Obsidian. I had this issue in my log and my classes kept forgetting their parents and excluded fields. After seeing several videos again I reinstalled it and I'm using it in a different vault with the same error.

I'm using Obsidian 1.87 (191} and my Metadata Menu is 0.8.7 and on mobile

Here is my error:

![Screenshot_20250313-221754](https://github.com/user-attachments/assets/4fb12201-a24a-461c-98d3-2576d0889d6c)
